From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
Date: Thu, 17 Apr 2025 16:10:53 +0200
Subject: GENIO: board: Fix ramdisk image overlapping

Current upstream kernel size is too big, resulting in overlapping areas:

    ERROR: RD image overlaps OS image (OS=0x45000000..0x480b0000)

Fix this by setting ramdisk image load address to 0x49000000.

Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
Signed-off-by: Louis-Alexis Eyraud <louisalexis.eyraud@collabora.com>
---
 include/configs/mt8188.h | 4 ++--
 include/configs/mt8195.h | 4 ++--
 include/configs/mt8365.h | 2 +-
 include/configs/mt8370.h | 4 ++--
 include/configs/mt8391.h | 2 +-
 5 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/include/configs/mt8188.h b/include/configs/mt8188.h
index 111111111111..222222222222 100644
--- a/include/configs/mt8188.h
+++ b/include/configs/mt8188.h
@@ -94,7 +94,7 @@
 	"fdtoverlay_addr_r=0x44c00000\0" \
 	"fdt_resize=0x3000\0" \
 	"kernel_addr_r=0x45000000\0" \
-	"ramdisk_addr_r=0x46000000\0" \
+	"ramdisk_addr_r=0x49000000\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE ".dtb\0" \
 	"splashimage=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"splashsource=mmc_fs\0" \
@@ -109,7 +109,7 @@
 	"fdtoverlay_addr_r=0x44c00000\0" \
 	"fdt_resize=0x3000\0" \
 	"kernel_addr_r=0x45000000\0" \
-	"ramdisk_addr_r=0x46000000\0" \
+	"ramdisk_addr_r=0x49000000\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE ".dtb\0" \
 	"splashimage=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"splashsource=sf\0" \
diff --git a/include/configs/mt8195.h b/include/configs/mt8195.h
index 111111111111..222222222222 100644
--- a/include/configs/mt8195.h
+++ b/include/configs/mt8195.h
@@ -133,7 +133,7 @@
 	"fdtoverlay_addr_r=0x44c00000\0" \
 	"fdt_resize=0x3000\0" \
 	"kernel_addr_r=0x45000000\0" \
-	"ramdisk_addr_r=0x46000000\0" \
+	"ramdisk_addr_r=0x49000000\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE ".dtb\0" \
 	"splashimage=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"splashsource=mmc_fs\0" \
@@ -148,7 +148,7 @@
 	"fdtoverlay_addr_r=0x44c00000\0" \
 	"fdt_resize=0x3000\0" \
 	"kernel_addr_r=0x45000000\0" \
-	"ramdisk_addr_r=0x46000000\0" \
+	"ramdisk_addr_r=0x49000000\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE ".dtb\0" \
 	"splashimage=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"splashsource=sf\0" \
diff --git a/include/configs/mt8365.h b/include/configs/mt8365.h
index 111111111111..222222222222 100644
--- a/include/configs/mt8365.h
+++ b/include/configs/mt8365.h
@@ -63,7 +63,7 @@
 	"fdtoverlay_addr_r=0x44c00000\0" \
 	"fdt_resize=0x3000\0" \
 	"kernel_addr_r=0x45000000\0" \
-	"ramdisk_addr_r=0x46000000\0" \
+	"ramdisk_addr_r=0x49000000\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE ".dtb\0" \
 	BOOTENV
 
diff --git a/include/configs/mt8370.h b/include/configs/mt8370.h
index 111111111111..222222222222 100644
--- a/include/configs/mt8370.h
+++ b/include/configs/mt8370.h
@@ -85,7 +85,7 @@
 	"fdtoverlay_addr_r=0x44c00000\0" \
 	"fdt_resize=0x3000\0" \
 	"kernel_addr_r=0x45000000\0" \
-	"ramdisk_addr_r=0x46000000\0" \
+	"ramdisk_addr_r=0x49000000\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE ".dtb\0" \
 	"splashimage=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"splashsource=mmc_fs\0" \
@@ -100,7 +100,7 @@
 	"fdtoverlay_addr_r=0x44c00000\0" \
 	"fdt_resize=0x3000\0" \
 	"kernel_addr_r=0x45000000\0" \
-	"ramdisk_addr_r=0x46000000\0" \
+	"ramdisk_addr_r=0x49000000\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE ".dtb\0" \
 	"splashimage=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"splashsource=sf\0" \
diff --git a/include/configs/mt8391.h b/include/configs/mt8391.h
index 111111111111..222222222222 100644
--- a/include/configs/mt8391.h
+++ b/include/configs/mt8391.h
@@ -100,7 +100,7 @@
 	"fdtoverlay_addr_r=0x44c00000\0" \
 	"fdt_resize=0x3000\0" \
 	"kernel_addr_r=0x45000000\0" \
-	"ramdisk_addr_r=0x46000000\0" \
+	"ramdisk_addr_r=0x49000000\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE ".dtb\0" \
 	"splashimage=" __stringify(CONFIG_SYS_LOAD_ADDR) "\0" \
 	"splashsource=mmc_fs\0" \
-- 
Armbian

